Confuser.Core/Utils.cs

```diff
@@ -189,6 +189,39 @@ public static class Utils {
 			}
 		}
 
+		public static int RemoveAndInsertRange<T>(this IList<T> self, int index, int removingCount, IEnumerable<T> collection) {
+			int previousCount = self.Count;
+			self.RemoveRange(index, removingCount);
+			self.InsertRange(index, collection);
+			return self.Count - previousCount;
+		}
+
+		public static void RemoveRange<T>(this IList<T> self, int index, int count) {
+			if (self is List<T> list) {
+				list.RemoveRange(index, count);
+				return;
+			}
+
+			for (int i = index + count - 1; i >= index; i--) {
+				self.RemoveAt(i);
+			}
+		}
+
+		public static void InsertRange<T>(this IList<T> self, int index, IEnumerable<T> collection) {
+			if (self is List<T> list && index < self.Count) {
+				list.InsertRange(index, collection);
+				return;
+			}
+
+			var localIndex = 0;
+			foreach (var element in collection) {
+				self.Insert(index + localIndex, element);
+				localIndex++;
+			}
+		}
+
+		public static void RemoveLast<T>(this List<T> self) => self.RemoveAt(self.Count - 1);
```

Comment:
The parameter here can be widened to `IList`, to match the other extension methods.
```suggestion
		public static void RemoveLast<T>(this IList<T> self) => self.RemoveAt(self.Count - 1);
```